Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverrideHandler.cpp

```diff
@@ -26,5 +28,35 @@
             }
             return false;
         }
+
+        bool PrefabOverrideHandler::RevertOverrides(AZ::Dom::Path path, LinkId linkId) const
+        {
+            PrefabSystemComponentInterface* prefabSystemComponentInterface = AZ::Interface<PrefabSystemComponentInterface>::Get();
+            if (prefabSystemComponentInterface != nullptr)
+            {
+                LinkReference link = prefabSystemComponentInterface->FindLink(linkId);
+                if (link.has_value())
+                {
+                    auto subTree = link->get().RemoveOverrides(path);
+                    if (subTree.IsEmpty())
+                    {
+                        AZ_Warning("Prefab", false, "PrefabOverrideHandler::RevertOverrides is called on a path that has no overrides.");
+                        return false;
+                    }
+
+                    ScopedUndoBatch undoBatch("Revert Prefab Overrides");
+                    PrefabUndoRevertOverrides* state = new Prefab::PrefabUndoRevertOverrides("Capture Override SubTree");
+                    state->Capture(path, AZStd::move(subTree), linkId);
+                    state->SetParent(undoBatch.GetUndoBatch());
+                    
+                    link->get().UpdateTarget();
+                    prefabSystemComponentInterface->PropagateTemplateChanges(link->get().GetTargetTemplateId());
+                    AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+                        &AzToolsFramework::ToolsApplicationRequestBus::Events::ClearDirtyEntities);
```

Comment:
It is to prevent any dirty entities from entering the CreateUndosForDirtyEntities() function when the scoped undo node is destroyed. At the end of this function, the only thing that should be modifying the prefab templates is the RevertOverrides() code itself. 

Although, all of this is prior to this PR: https://github.com/o3de/o3de/pull/13127 . So I'll try to remove that and see if it is still needed.

Reply:
Removed this as it no longer looks like its is needed. Probably fixed by the above PR